### PR TITLE
daml-on-sql + sandbox + sandbox-classic: Split out custom CLI parsing.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
I'm not sure this is the cleanest approach, but this is an attempt to parse CLI arguments slightly differently depending on the version of Sandbox we're using. For example, we can't disable implicit party allocation in Sandbox Classic, so it shouldn't be an option, and Sandbox Next doesn't support scenarios, so they shouldn't be around.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/daml-on-sql/src/main/scala/on/sql/Cli.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import com.daml.ledger.participant.state.v1.SeedService.Seeding
+import com.daml.platform.common.LedgerIdMode
+import com.daml.platform.sandbox.cli.{CommonCli, SandboxCli}
+import com.daml.platform.sandbox.config.SandboxConfig
+import scopt.OptionParser
+
+private[sql] final class Cli(override val defaultConfig: SandboxConfig = DefaultConfig)
+    extends SandboxCli {
+
+  private val seedingMap = Map[String, Option[Seeding]](
```
[the remark this answers is not in the document]
Comment (SamirTalwar-DA) [reply ledger/daml-on-sql/src/main/scala/on/sql/Cli.scala:15]: Two reasons.

1. We use weak seeding in our conformance tests.
2. Just like us, I expect users of DAML-on-SQL to run end-to-end tests. Perhaps lots of them, in parallel, with each one running against a fresh ledger. In this event, strong seeding might block, leading to timeouts that are very hard to diagnose.

There's a reason the option is called "testing-weak". It's for testing. 🙂

[the remark this answers is not in the document]
Comment (gerolf-da) [reply ledger/daml-on-sql/src/main/scala/on/sql/Cli.scala:15]: already. Can I then still interest you in removing `testing-static`? This is really only useful for reproducible contract ids for demos or manuals.

[the remark this answers is not in the document]
Comment (SamirTalwar-DA) [reply ledger/daml-on-sql/src/main/scala/on/sql/Cli.scala:15]: Sold! I also pulled out the common logic for seeding mode options.

**Comment by SamirTalwar-DA** [comment]
Just realized the tests don't run on this PR because it's to be merged into PR #6829, so you'll have to trust me when I say I ran them all.

**Pull request merged**